packages/addon-dev/src/rollup-hbs-plugin.ts

```diff
@@ -1,22 +1,64 @@
 import { createFilter } from '@rollup/pluginutils';
-import type { Plugin } from 'rollup';
 import { readFileSync } from 'fs';
 import { hbsToJS } from '@embroider/shared-internals';
+import path from 'path';
+import { pathExists } from 'fs-extra';
+
+import type { Plugin } from 'rollup';
 
 export default function rollupHbsPlugin(): Plugin {
   const filter = createFilter('**/*.hbs');
 
   return {
     name: 'rollup-hbs-plugin',
     async resolveId(source: string, importer: string | undefined, options) {
-      const resolution = await this.resolve(source, importer, {
+      let resolution = await this.resolve(source, importer, {
         skipSelf: true,
         ...options,
       });
 
-      const id = resolution?.id;
+      let id = resolution?.id;
 
-      if (!filter(id)) return null;
+      // if id is undefined, it's possible we're importing a file that that rollup
+      // doesn't natively support such as a template-only component that the author
+      // doesn't want to be available on the globals resolver
```

Comment:
This comment is misleading. The globals resolver isn't really relevant to this. You control global resolving by deciding whether or not to put a module into appReexports. But this code runs for your modules whether or not they're in appReexports.